### PR TITLE
Scope Zookeeper & Kafka fixtures to session

### PR DESCRIPTION
**Pull request by dagardner-nv**
## Description
* `pytest --run_slow --fail_missing --run_kafka` 2 minutes faster
* `kafka_consumer` & `kafka_topics` fixtures are still function scoped, ensuring each test will receive unique topics & client
* Move Kafka related testing code to `tests/_utils/kafka.py`

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: tests/_utils/kafka.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+PARTITION_ASSIGNMENT_TIMEOUT = 15
+
+# actual topic names not important, but we will need two of them.
+KafkaTopics = namedtuple('KafkaTopics', ['input_topic', 'output_topic'])
+
+# pylint: disable=invalid-name
+_kafka_consumer = None
+kafka_server = None
+zookeeper_proc = None
+# pylint: enable=invalid-name
```
Comment (cwharris) [inline tests/_utils/kafka.py:39]: what's with the pylint directives? aside from the `_kafka_consumer`, these look pretty normal?

File: tests/_utils/kafka.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+
+@pytest.fixture(name='kafka_topics', scope='function')
+def kafka_topics_fixture():
+    yield KafkaTopics(f'morpheus_unittest_input_{time.time()}', f'morpheus_unittest_output_{time.time()}')
```
Comment (cwharris) [inline tests/_utils/kafka.py:44]: Is `time` the best discriminator we can use? This has the potential to create a lot of topics if we run tests repeatedly without restarting kafka. What are your thoughts on using something like this?
```python
import os
import re
def get_test_name_for_kafka_topic():
    test_name = os.environ.get("PYTEST_CURRENT_TEST")
    test_name = re.sub("[:/\[\]\(\)]", "-", test_name)
    test_name = re.sub(",", "_", test_name)
    return test_name
```
which produces something along these lines:
```python
"tests-test_kafka_oauth.py--test_kafka_oauth-use_cpp- -call-"
"tests-test_kafka_oauth.py--test_kafka_oauth-use_python- -call-"
```

File: tests/_utils/kafka.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+    producer = KafkaProducer(bootstrap_servers=bootstrap_servers, client_id=client_id)
+    for row in data:
+        if isinstance(row, dict):
+            row = json.dumps(row)
+        producer.send(kafka_topic, row.encode('utf-8'))
+        num_records += 1
+
+    producer.flush()
+
+    assert num_records > 0
```
Comment (cwharris) [inline tests/_utils/kafka.py:133]: Why are we summing `num_records`? Wouldn't that be the same as `len(data)`? Unless `producer.send` or `json.dumps` raises an error... in which case we wouldn't use the value anyway?

File: tests/_utils/kafka.py [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    return num_records
+
+
+def write_file_to_kafka(bootstrap_servers: str,
+                        kafka_topic: str,
+                        input_file: str,
+                        client_id: str = 'morpheus_unittest_writer') -> int:
+    """
+    Writes data from `inpute_file` into a given Kafka topic, emitting one message for each line int he file.
+    Returning the number of messages written
```
Comment (cwharris) [inline tests/_utils/kafka.py:144]:
```suggestion
    Writes data from `input_file` into a given Kafka topic, emitting one message for each line in the file, 
    and returns the number of messages written.
```

**Review by cwharris — APPROVED** [review]
Thanks for consolidating all of the kafka utils to the appropriate file. :pray:

File: tests/_utils/kafka.py (hunk 1 above)
Comment (dagardner-nv) [reply tests/_utils/kafka.py:39]: Pylint believes these to be constants and wants an upper-case name.

File: tests/_utils/kafka.py (hunk 3 above)
Comment (dagardner-nv) [reply tests/_utils/kafka.py:133]: I don't recall how we ended up with this code. I didn't modify this method, simply moved it from another module, however I believe you are correct num_records should be the same as `len(data)`

File: tests/_utils/kafka.py (hunk 1 above)
Comment (cwharris) [reply tests/_utils/kafka.py:39]: weird!

**Review by mdemoret-nv — APPROVED** [review]

**Pull request merged**